### ISSUE TITLE
Ideas

### ISSUE THREAD

**Issue by leoheck** [scantailor/scantailor#258]
I have a poor scanned but searchable/selectable pdf text file. Scantailor made the manuscript awesome. But the problem is that I lost the possibility to select/search inside the document. 

Someone knows any way to make the document searchable/selectable again? Maybe an OCR tool can help with this. Also, It would be awesome If scantailor could export the work as pdf, and scan the text and enabling a way to add an index to the pdf.

**Comment by boredland** [comment]
You've got two options now:
1.a) Extract your old text-layer 
1.b) Do another text recognition
2. and add it as an overlay to your optimized scans.

For 1 a) You can use pdf2html: http://www.tutorialspoint.com/unix_commands/pdftohtml.htm Keep in mind that you'd like all the pages to be in separated html-files for pdfbeads, so you'd probably want to create a bash-script that iterates -f and -l to each page of your document. 
For 1 b) I'd choose tesseract. It has a lot of languages supported and is available in all linux distributions
For 2. pdfbeads is the best tool around, featuring jbig2-compression etc.

**Comment by Tiesenhausen** [comment]
A simple way to deal with this is with a pdf program that includes an OCR option.
I use PDF-Exchange Editor (free download). Open your awesome Scantailor-processed pdf with it, select Documents and then OCR pages..., sit back and wait a few minutes. It handles English, French, German and Spanish.
PDF-exchange Editor also has an Export function to create a jpg for each page of a pdf, to make the raw material for ScanTailor to assemble into an awesome new document. It's my default pdf program for all purposes, works splendidly.
Michael Bates

**Comment by leoheck** [comment]
@boredland and @Tiesenhausen thank you very mutch for the answers. I will try your ideas to see what works better for me. Now I am thinking that we can add an extra section to the wiki page explaining or referencing a way to do this process. What do you think?

**Comment by boredland** [comment]
Sounds like a good idea. Nevertheless: I shouldn't be that big of a problem to integrate pdf-creation into a future release, if I remember it correctly, it has been there in some earlier featured-release.

**Comment by leoheck** [comment]
Yeah, or at least supply some initial scripts to do that and a wiki page explaining the process. Then after some discussion, eventually, this can be integrated on the `scantailor`.

Also, ``html2pdf`` didn't work for me. It extracted only images. I don't know why since the ugly original pdf has the text. ``tesseract`` just worked very good. I think this is because the scan is good and the ``scantailor`` did such a great job. 

I still working to use the ``pdfbeads`` right now.

**Comment by leoheck** [comment]
I have some issues. 

Some definitions to easily discussion:
- `orig.pdf` (original, ugly, with text)
- `new.pdf` (new, good-looking, with text layer overlay)

1. I figured out that `pdfbeads` can use the `orig.pdf` as text input. The problem is that `scantailor` could change DPI and text positions with reference to the page. So, this isn't a good solution as the text layer will not match the text on the background image. I also didn't find a way to fix this and it will be tedious since text position can be different for each page.
2. How to generate `html` from `tesseract`? The `hocr` throw me an error. Using this `-c tessedit_create_hocr=1 -c tessedit_pageseg_mode=1` instead. Now I have an `html`-like file with `hocr` extension which I can rename to `html` if it is required.
3. How to use `tif` (from `scantailor`) with `html` files to make `pdfbead` generate the `new.pdf`? I put all `tif` and `html` (and `hocr`) files on the same folder but `pdfbead` does not generate a text layer. How can I validate if the layer is there? How can I know if the text will be added to the right position?

**Comment by boredland** [comment]
Hi @leoheck,

tesseract itself has been able to generate pdfs with a text-layer, so this shouldn't be the problem. I think the much more severe issue is, that pdfbeads is old and never worked solid. 
I think: In the end, the jbig2enc was created to compress image-layers into jbig2-pdfs. 
Perhaps this could be the first step: adding jbig2-pdfs as an output-option. 
Since tesseract already has a pdf-output I think that would be the place where we could have that the cheapest. 
A future workflow:
1. Prepare images via Scantailor.
2. Create Single-Page PDFs using tesseract with jbig2-compression.
3. Merge those pdfs without recompression (like pdftk).

I am sure, that would give you the best results in OCR and Compression, OpenSource could do at the moment. The only thing missing right now is the jbig2-compression in tesseract. Perhaps you can create an issue there? I'd really like to look into that next month.

**Comment by boredland** [comment]
Nevermind, created an issue, so we could hear from them: 
https://github.com/tesseract-ocr/tesseract/issues/726

**Comment by leoheck** [comment]
A new idea for improving scantailor flow is to start the project from another pdf to reduce the effort in creating separated images.

**Comment by atla5** [comment]
@leoheck: similar issues/solutions tend to pop up here every once in awhile. would making an open repository for commonly-used/useful scripts be helpful?

**Comment by leoheck** [comment]
Yeah, I think this could be a good starting point for people like me that would like to automate the whole process. I created some scripts. We can discussion how to organize this before. What do you think? Do you have any ideas to share?
